test/success/num.prl

```diff
@@ -22,7 +22,7 @@ Thm Pred : [
   lam a.
   fresh a', ind -> elim a;
   [ `zero ];
-  [ use a' ]
+  [ `,a' ]
```

Comment:
@favonia Good question; I think it would be cool if `use` could "just work" in cases where the kinds are not the same, but compatible in the appropriate way. But you know better than I how feasible that is.